internal/provider/provider_test.go

```diff
@@ -143,18 +140,52 @@ func testAccPreCheck(t *testing.T) {
 			t.Fatalf("%s must be set for acceptance tests", JujuCACertEnvKey)
 		}
 	}
+	confResp := configureProvider(t, Provider, false)
+	TestClient = confResp.ResourceData.(*juju.Client)
+}
+
+func configureProvider(t *testing.T, p provider.Provider, expectFail bool) provider.ConfigureResponse {
+	schemaResp := provider.SchemaResponse{}
+	Provider.Schema(context.Background(), provider.SchemaRequest{}, &schemaResp)
+	assert.Equal(t, schemaResp.Diagnostics.HasError(), false)
+
+	conf := jujuProviderModel{
+		ControllerAddrs: types.String{},
+		UserName:        types.String{},
+		Password:        types.String{},
+		CACert:          types.String{},
+	}
 
-	err := Provider.Configure(context.Background(), terraform.NewResourceConfigRaw(nil))
-	if err != nil {
-		t.Fatal(err)
+	mapTypes := map[string]attr.Type{
+		JujuController: types.StringType,
+		JujuUsername:   types.StringType,
+		JujuPassword:   types.StringType,
+		JujuCACert:     types.StringType,
 	}
+
+	val, confObjErr := types.ObjectValueFrom(context.Background(), mapTypes, conf)
+	assert.Equal(t, confObjErr.HasError(), false)
+
+	tfval, tfValErr := val.ToTerraformValue(context.Background())
+	assert.Equal(t, tfValErr, nil)
+
+	c := tfsdk.Config{Schema: schemaResp.Schema, Raw: tfval}
+	confReq := provider.ConfigureRequest{Config: c}
+	confResp := provider.ConfigureResponse{Diagnostics: diag.Diagnostics{}}
+
+	p.Configure(context.Background(), confReq, &confResp)
+	if !expectFail {
+		assert.Equal(t, confResp.Diagnostics.HasError(), false)
+	}
```

Comment:
I dislike double negatives in code. You could do something like:

```suggestion
	assert.Equal(t, confResp.Diagnostics.HasError(), expectFail)
```

Reply:
Yeah I don't really like doing this here anyways, I'll remove this and let all the tests that expect this to fail to check that in their own code 👍